docs/guides/latest/volumesnapshot/deployment.md

```diff
@@ -0,0 +1,513 @@
+      - metadata:
+          name: restore-data
+        spec:
+          accessModes: [ "ReadWriteOnce" ]
+          storageClassName: "standard"
+          resources:
+            requests:
+              storage: 1Gi
+          dataSource:
+            kind: VolumeSnapshot
+            name: source-data-1563171247
+            apiGroup: snapshot.storage.k8s.io
+      - metadata:
+          name: restore-config
+        spec:
+          accessModes: [ "ReadWriteOnce" ]
+          storageClassName: "standard"
+          resources:
+            requests:
+              storage: 1Gi
+          dataSource:
+            kind: VolumeSnapshot
+            name: source-config-1563171247
+            apiGroup: snapshot.storage.k8s.io
+```
+
+Here,
+
+- `spec.target.volumeClaimTemplates`:
+  - `metadata.name` is the name of the restored `PVC`.
+  - `spec.dataSource`: `spec.dataSource` specifies the source of the data from where the newly created PVC will be initialized. It requires the following fields to be set:
+    - `apiGroup` is the group for resource being referenced. Now, Kubernetes supports only `snapshot.storage.k8s.io`.
+    - `kind` is resource of the kind being referenced. Now, Kubernetes supports only `VolumeSnapshot`.
+    - `name` is the `VolumeSnapshot` resource name. In `RestoreSession` crd, You must set the VolumeSnapshot name directly.
+
+Let's create the `RestoreSession` crd we have shown above.
+
+```console
+$ kubectl create -f ./docs/examples/guides/latest/volumesnapshot/deployment/restoresession.yaml
+restoresession.stash.appscode.com/restore-pvc created
+```
+
+Once, you have created the `RestoreSession` crd, Stash will create a job to restore. We can watch the `RestoreSession` phase to check if the restore process is succeeded or not.
+
+Run the following command to watch RestoreSession phase,
+
+```console
+$ watch -n 1 kubectl get restore -n demo
+Every 1.0s: kubectl get restore -n demo                      suaas-appscode: Tue Jun 18 18:35:41 2019
+
+NAME          REPOSITORY-NAME   PHASE       AGE
+restore-pvc                     Running     10s
+restore-pvc                     Succeeded   1m
+```
+
+So, we can see from the output of the above command that the restore process succeeded.
+
+**Verify Restored PVC :**
+
+Once a restore process is complete, we will see that new PVCs with the name `source-data` and `source-config ` has been created.
+
+Verify that the PVCs has been created by the following command,
+
+```console
+$ kubectl get pvc -n demo
+NAME             STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
+restore-config   Bound    pvc-26758eda-a6ca-11e9-9f3a-42010a800050   1Gi        RWO            standard       30s
+restore-data     Bound    pvc-267335ff-a6ca-11e9-9f3a-42010a800050   1Gi        RWO            standard       30s
+```
+
+Notice the `STATUS` field. `Bound` indicates that PVC has been initialized from the respective VolumeSnapshot.
```

Comment:
Notice the `STATUS` field. It indicates that the respective PV has been provisioned and initialized from the respective VolumeSnapshot by CSI driver and the PVC has been bound with the PV.